File: flatseeker-ui/src/app/model/dict/ProcessStatusDict.ts
export class ProcessStatusDict {
  public key: string;
  public title: string;
  public color: string;
  public active: boolean;

  constructor(key: string, title: string, color: string, active: boolean) {
    this.key = key;
    this.title = title;
    this.color = color;
    this.active = active;
  }
}

File: flatseeker-backend/src/main/java/com/yakzhanov/flatseeker/model/dto/dict/ProcessStatusDict.java
package com.yakzhanov.flatseeker.model.dto.dict;

import com.yakzhanov.flatseeker.model.ProcessStatus;
import lombok.Data;

@Data
public class ProcessStatusDict {

    private String key;
    private String title;
    private String color;

    public ProcessStatusDict(ProcessStatus status) {
        this.key = status.name();
        this.title = status.getTitle();
        this.color = status.getColor();
    }
}
